File: 0563_binary_tree_tilt/find_tilt.java
public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

class Solution {
    
    int res;
    
    // Time: O(n)
    // Space: O(n)
    public int findTilt(TreeNode root) {
        res = 0;
        helper(root);
        return res;
    }
    
    private int helper(TreeNode node) {
        if (node == null) {
            return 0;
        }
        int l = helper(node.left);
        int r = helper(node.right);
        int tilt = Math.abs(l - r);
        res += tilt;
        return l + node.val + r;
    }
}

File: 0563_binary_tree_tilt/find_tilt.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


def findTilt(root: TreeNode) -> int:
    """
    Time: O(n)
    Space: O(n)
    """
    res = 0
    def helper(node: TreeNode) -> int:
        nonlocal res
        if not node:
            return 0
        l, r = helper(node.left), helper(node.right)
        tilt = abs(l - r)
        res += tilt
        return l + node.val + r
    helper(root)
    return res